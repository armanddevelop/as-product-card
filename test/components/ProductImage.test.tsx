import React from 'react';
import render from "react-test-renderer"
import ProductImage from '../../src/components/ProductImage';
import {ProductCard} from '../../src/components/ProductCard';
import { product2 } from '../data/products';

describe("Product image component UT",()=>{
  test("Test the image is send to the component",()=>{
    const wrapper = render.create(
      <>
        <ProductCard product={product2}>
          {
            ()=>(<ProductImage className='test' img={product2.img} style={{color:"green"}}/>)
          }
        </ProductCard>
      </>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe("Product image component UT",()=>{
  test("Test the image is no send to the component",()=>{
    const wrapper = render.create(
      <>
        <ProductCard product={product2}>
          {
            ()=>(<ProductImage className='test' img='https://alicha.jpg' style={{color:"green"}}/>)
          }
        </ProductCard>
      </>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});