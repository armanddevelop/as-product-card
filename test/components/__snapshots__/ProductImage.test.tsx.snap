// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product image component UT Test the image is no send to the component 1`] = `
<div
  className="productCard undefined"
>
  <img
    className="productImg test"
    src="https://alicha.jpg"
    style={
      Object {
        "color": "green",
      }
    }
  />
</div>
`;

exports[`Product image component UT Test the image is send to the component 1`] = `
<div
  className="productCard undefined"
>
  <img
    className="productImg test"
    src="./coffee-mug.png"
    style={
      Object {
        "color": "green",
      }
    }
  />
</div>
`;
